shared/utils/typedefs-wrappers.hpp

```diff
@@ -563,6 +565,122 @@ struct SafePtrUnity : public SafePtr<T, true> {
     }
 };
 
+template <typename T>
+#ifdef HAS_CODEGEN
+    requires(std::is_convertible_v<T*, UnityEngine::Object*>)
+#elif !defined(NO_CODEGEN_USE)
+    requires(std::is_convertible_v<T*, Il2CppObject*>)
+#endif
+struct UnityW {
+    UnityW() = default;
+    UnityW(T* t) : innerPtr(t) {}
+
+    template <typename U>
+    requires(std::is_convertible_v<U*, T*>)
+    UnityW(UnityW<U> u) : innerPtr(u.innerPtr) {}
+    
+    UnityW(void* p) : innerPtr(reinterpret_cast<T*>(p)) {}
```

Comment:
void* ctor should be explicit, and also it can be constexpr if you use static_cast

Reply:
actually try to make these all constexpr where possible

Reply:
actually I did it myself but I can't push to this branch so I'll just merge it and manually add my changes